packager/media/codecs/h264_byte_to_unit_stream_converter_unittest.cc

```diff
@@ -79,5 +79,50 @@ TEST(H264ByteToUnitStreamConverter, ConversionFailure) {
   EXPECT_FALSE(converter.GetDecoderConfigurationRecord(&decoder_config));
 }
 
+TEST(H264ByteToUnitStreamConverter, NaluConversionWithSpsExtension) {
+  const uint8_t kSpsExtArr[] = {
+      0x00, 0x00, 0x00, 0x01,	// Start code
+      0x09, 			// AUD Type
+      0xF0, 			// Primary pic type
+      0x00, 0x00, 0x00, 0x01, 	// Start code
+      // Some SPS data
+      0x67, 0x64, 0x00, 0x1E, 0xAC, 0xD9, 0x40, 0xB4, 
+      0x2F, 0xF9, 0x7F, 0xF0, 0x00, 0x80, 0x00, 0x91, 
+      0x00, 0x00, 0x03, 0x03, 0xE9, 0x00, 0x00, 0xEA, 
+      0x60, 0x0F, 0x16, 0x2D, 0x96, 
+      0x00, 0x00, 0x00, 0x01, 	// Start code
+      // Some PPS data
+      0x68, 0xFE, 0xFD, 0xFC, 0xFB, 0x11, 0x12, 0x13, 0x14, 0x15, 
+      0x00, 0x00, 0x00, 0x01,	// Start code
+      // Some SPS Extension data
+      0x6D, 0x33, 0x01, 0x57, 0x78, 
+      0x00, 0x00, 0x00, 0x01, 	// Start code
+      // Input NALU
+      0x06, 
+      0xFD, 0x78, 0xA4, 0xC3, 0x82, 0x62, 0x11, 0x29, 0x77, 
+      0x00, 0x00, 0x00, 0x01, 	// Start code
+      // SPS Data
+      0x00, 0x02, 0x67, 0x64, 
+      0x00, 0x00, 0x00, 0x01, 	// Start code
+      // PPS Data
+      0x00, 0x01, 0x68, 0xFE, 
+      0x00, 0x00, 0x00, 0x01, 	// Start code
+      // SPS Extension data
+      0x00, 0x05, 0x6d, 0x33, 0x01, 0x57, 0x78
+  };
+
+  int kSpsExtSize = arraysize(kSpsExtArr);
+  std::vector<uint8_t> kSpsExt(kSpsExtArr, kSpsExtArr + kSpsExtSize);
+  H264ByteToUnitStreamConverter converter(
+      H26xStreamFormat::kNalUnitStreamWithParameterSetNalus);
+
+  std::vector<uint8_t> unit_stream_output;
+  ASSERT_TRUE(converter.ConvertByteStreamToNalUnitStream(
+              kSpsExt.data(), kSpsExt.size(),
+              &unit_stream_output));                                                 
```

Comment:
Can you define: uint8_t kExpectedUnitStream and add an expectation:

```c++
  EXPECT_EQ(std::vector<uint8_t>(std::begin(kExpectedUnitStream), 
                                 std::end(kExpectedUnitStream)),	
            unit_stream);
```

Reply:
done, although I used slightly different syntax for EXPECT_EQ(), following an example from the nalu unittest